examples/monitor-prom-remotewrite-enable-basic-auth/README.md

```diff
@@ -0,0 +1,151 @@
+  ```
+  
+  The output is similar to this:
+  
+  ```bash
+        NAME                                       TIME
+    alertmanagers.monitoring.coreos.com            2020-04-12T14:22:49Z
+    podmonitors.monitoring.coreos.com              2020-04-12T14:22:50Z
+    prometheuses.monitoring.coreos.com             2020-04-12T14:22:50Z
+    prometheusrules.monitoring.coreos.com          2020-04-12T14:22:50Z
+    servicemonitors.monitoring.coreos.com          2020-04-12T14:22:51Z
+  ```
+
+- Has Ingress nginx controller installed. [Doc](https://kubernetes.github.io/ingress-nginx/)
+
+  This could by verified by the following command:
+
+  ```bash
+  >  kubectl get pods -n ingress-nginx 
+  ```
+
+  The output is similar to this:
+
+  ```bash
+        NAME                                       TIME
+  ingress-nginx-admission-create-8ljj2       0/1     Completed   0          79m
+  ingress-nginx-admission-patch-mcn8r        0/1     Completed   0          79m
+  ingress-nginx-controller-cb87575f5-sd6kb   1/1     Running     0          79m
+
+  ``` 
+## Initialize
+
+### Initialize remote write prometheus
+```bash
+> kubectl -n <namespace> apply -f ./prometheus.yaml
+> kubectl -n <namespace> apply -f ./prometheus-service.yaml
+```
+
+Wait for Initialize remote write prometheus done:
+
+```bash
+$ kubectl get pods -n <namespace>| grep prometheus-basic
+NAME         AGE
+prometheus-basic-0                 2/2     Running   0          83m
+```
+
+### Create auth ingress
+Ref [Doc](https://kubernetes.github.io/ingress-nginx/examples/auth/basic/).
+
+First step: create username and password.
+```bash
+> htpasswd -c auth admin
+New password: <admin>
+New password:
+Re-type new password:
+Adding password for user admin
+```
+
+Second step: create ingress.
+```bash
+> kubectl apply -f ./ingress.yaml -n <namespace>
+```
+
+Third step: validate auth.
+```bash
+> kubectl port-forward svc/ingress-nginx-controller 8080:80 -n ingress-nginx
+Use curl with the correct credentials to connect to the ingress
+> curl -v  http://127.0.0.1:8080 -H 'Host: test.prometheus.com' -u 'admin:admin'
+*   Trying 127.0.0.1...
+* TCP_NODELAY set
+* Connected to 127.0.0.1 (127.0.0.1) port 8080 (#0)
+* Server auth using Basic with user 'admin'
+> GET / HTTP/1.1
+> Host: test.prometheus.com
+> Authorization: Basic YWRtaW46YWRtaW4=
+> User-Agent: curl/7.64.1
+> Accept: */*
+> 
+< HTTP/1.1 302 Found
+< Date: Sun, 13 Mar 2022 17:44:12 GMT
+< Content-Type: text/html; charset=utf-8
+< Content-Length: 29
+< Connection: keep-alive
+< Location: /graph
+< 
+<a href="/graph">Found</a>.
+
+* Connection #0 to host 127.0.0.1 left intact
+* Closing connection 0
+```
+
+### Create tidbMonitor and TiDB
+
+Create TiDB:
+```bash
+> kubectl apply -f tidb-cluster.yaml -n <namespace>
+```
+
+Create auth secret and TiDBMonitor:
+```bash
+> kubectl apply -f basic-auth.yaml -n <namespace>
+> kubectl apply -f tidb-monitor.yaml -n <namespace>
+```
+
+Add hostAliases to tidbmonitor statefulset:
```

Comment:
This change will be reverted by the TiDB Operator, right?

Reply:
Can we just exec into the Pod to change the /etc/hosts temporarily?

Reply:
<img width="364" alt="image" src="https://user-images.githubusercontent.com/13257127/160321472-b4cc8ecc-7d8a-4f99-9c41-25b3c5be5685.png">

we  can't change /etc/hosts temporarily.